__tests__/app.test.js

```diff
@@ -15,3 +23,33 @@ describe('GET /api', () => {
       });
   });
 });
+
+describe('GET /api/topics', () => {
+  test('200: Responds with an array of topic objects containing slug and description properties', () => {
+    return request(app)
+      .get('/api/topics')
+      .expect(200)
+      .then(({ body: { topics } }) => {
+        expect(topics.length).toBe(3);
```

Comment:
This is a really important test! Do you know why?

Reply:
My understanding is that if we write an assertion about received object properties using .forEach, and our test receives an empty array, the test will pass even if the array is empty!

I read [this article](https://macarthur.me/posts/looping-over-empty-array-items/) about how sparse arrays and .forEach can have some surprising interactions, so, if we assert the length of our array before doing any checking of properties, we know our .forEach is doing what it's intended to?

Reply:
Exactly that 🥇 Empty arrays are slightly more damning than sparse arrays. Even if we used a for loop with an empty array: `for ( let i = 0; i < array.length; i++)` it'll first create the variable `i`, then check `i < array.length` which will check `0 < 0` which is `false` so the loop will not run (even though it would run with a sparse array).

Reply:
Oh, that's such an interesting interaction, okay!! I think visualising it as a `for` loop like that makes it really easy to understand, I'll keep that in mind, thank you!